views/my_packages.register.dt

```diff
@@ -15,13 +15,13 @@ block body
 	.inputForm
 		h1 Add new package
 			p.light
-				a.blind(href="#{req.rootDir}publish") Learn more 
+				a.blind(href="#{req.rootDir}publish") Learn more
 				| about publishing.
 
 		form(method="POST", action="#{req.rootDir}register_package")
 			p
 				label(for="project") Repository URL
-				input(type="text", name="url", value=url, pattern="^(https?://)?(www\.)?(github(\.com)?|gitlab(\.com)?|bitbucket(\.org)?)/[^/]+/[^/]+([?#.].*)?$", placeholder="github.com/user/repository")
+				input(type="text", name="url", value=url, pattern="^(https?:\/\/)?(www\.)?(github(\.com)?|gitlab(\.com)?|bitbucket(\.org)?)(/[^/]+)+/[^/]+([?#.].*)?$", placeholder="github.com/user/repository")
```

Comment:
This allows invalid patterns for GitHub & Bitbucket, e.g. github.com/foo/bar/barrrr

Reply:
Yes, it will. The right regex is:

    ^(https?:\/\/)?(www\.)?(((github(\.com)?|bitbucket(\.org)?)\/[^/]+\/[^/]+([?#.].*)?)|(gitlab(\.com)?(\/[^/]+)+\/[^/]+([?#.].*)?))$

I'll put it in the code tomorrow and commit the fix. In the meantime I would be grateful if anyone tested it once again to confirm its correctness, as it gets long now.